flocker/node/agents/gce.py

```diff
@@ -254,22 +271,23 @@ def _dataset_id_to_blockdevice_id(dataset_id):
 
 def _extract_attached_to(disk):
     """
-    Given a GCE disk resource, determines the unicode name of the machine that
-    it is attached to.
+    Given a GCE disk resource, determines the unicode name of the
+    machine that it is attached to. If the disk is attached READ_ONLY
+    to multiple machines, we simply return the first instance (flocker
+    doesn't support volumes attached to multiple machines).
```

Comment:
In that case, I think it'd be better to raise an exception in that case. And / or log an error.

Elsewhere we raise a `ValueError` in this case.
 * https://clusterhq.atlassian.net/browse/FLOC-1854
 * https://github.com/ClusterHQ/flocker/blob/master/flocker/node/agents/cinder.py#L736